src/cpp/CMakeLists.txt

```diff
@@ -117,11 +117,11 @@ set(${PROJECT_NAME}_source_files
     fastdds/domain/DomainParticipant.cpp
     fastdds/domain/qos/DomainParticipantQos.cpp
     fastdds/domain/qos/DomainParticipantFactoryQos.cpp
-    fastdds/builtin/typelookup/common/TypeLookupTypes.cpp
-    fastdds/builtin/common/RPCHeadersImpl.cpp
-    fastdds/builtin/typelookup/TypeLookupManager.cpp
-    fastdds/builtin/typelookup/TypeLookupRequestListener.cpp
-    fastdds/builtin/typelookup/TypeLookupReplyListener.cpp
+    fastdds/builtin/typelookupservice/common/TypeLookupTypes.cpp
+    fastdds/builtin/typelookupservice/common/RPCHeadersImpl.cpp
+    fastdds/builtin/typelookupservice/TypeLookupManager.cpp
+    fastdds/builtin/typelookupservice/TypeLookupRequestListener.cpp
+    fastdds/builtin/typelookupservice/TypeLookupReplyListener.cpp
```

Comment:
If we are going to rename the folder, then I suggest the following name: `type_lookup_service`